### ISSUE TITLE
Argument validation should be more strict.

### ISSUE THREAD

**Issue by dingjs** [minio/minio-java#651]
Hi!
In this method:
``` java
public void removeObject(String bucketName, String objectName)
```
If objectName is null or empty(due to some bad coding),this method will delete the whole bucket,which is very dangerous,even removeBucket would not recursively delete objects.

The api description should be more precise,and do the argument validation strictly,if there is a  invalid argument , an IllegalArgumentException would be better.

There is other  method like `getObject(String bucketName, String objectName)`,if objectName is null or empty,this method would act as ListBucketResult with no prefix.

### 3.0.8

**Comment by nitisht** [comment]
Thanks for reporting @dingjs we'll send a fix soon

**Comment by nitisht** [comment]
@dingjs this is fixed with #652 .
